src/error.rs

```diff
@@ -49,7 +52,7 @@ impl Error {
 		if status == ::hyper::status::StatusCode::TooManyRequests {
 			if let Some(Value::Object(ref map)) = value {
 				if let Some(delay) = map.get("retry_after").and_then(|v| v.as_u64()) {
-					return Error::RateLimited(delay)
+					return Error::RateLimited(delay);
```

Comment:
Well here's my spin on it:

Absence of a semicolon is pretty much only used for implicit return statements, so much so that rustc uses the presence of a semicolon on the last line with no return statement to hint to the user  that they probably meant to not have the semicolon.  The [rust style guide](https://github.com/rust-lang-nursery/fmt-rfcs/blob/master/guide/guide.md#expressions-in-statement-position) indicates that you should use a semicolon in this situation.

Ultimately though I think the whole decision is entirely arbitrary and pointless.  Both versions produce the exact same assembly code, and communicate the exact same intent and semantics.